kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ClusterConfigManager.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.controllers;
+
+import io.confluent.kafkarest.entities.ClusterConfig;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public interface ClusterConfigManager {
+
+  /**
+   * Returns the list of Kafka {@link ClusterConfig ClusterConfigs}.
+   */
+  CompletableFuture<List<ClusterConfig>> listClusterConfigs(
+      String clusterId, ClusterConfig.Type type);
+
+  /**
+   * Returns the Kafka {@link ClusterConfig} with the given {@code name}.
+   */
+  CompletableFuture<Optional<ClusterConfig>> getClusterConfig(
+      String clusterId, ClusterConfig.Type type, String name);
+
+  /**
+   * Creates or updates the Kafka {@link ClusterConfig} with the given {@code name} with the given
+   * {@code newValue}.
+   */
+  CompletableFuture<Void> upsertClusterConfig(
```

Comment:
nit : `upsertClusterConfig` -> `updateClusterConfig`

Reply:
update: I think upsert is good too, I had not come across it much.

Reply:
Cluster configs are different from normal configs in that they do not already exist. For broker and topic configs, all of them already exists from the start, so you can only update a config value. For cluster configs, at the beginning you actually have no configs. When you update a value for a config that does not exist, it will create one. KIP-226 is confusing. In any case, upsert is the usual name for when you are doing a update-or-create operation.